py/dml/structure.py

```diff
@@ -1398,9 +1411,9 @@ def process_method_implementations(obj, name, implementations,
                 typecheck_method_override(impl.method_ast,
                                           overridden.method_ast,
                                           location)
-            if (impl.method_ast.site.dml_version() == (1, 2) and throws
+            if (impl.site.dml_version() == (1, 2) and throws
                 and not overridden.throws
-                and overridden.method_ast.site.dml_version() == (1, 4)):
+                and overridden.site.dml_version() == (1, 4)):
```

Comment:
The refactoring for this region to use `impl.site` instead of `impl.method_ast.site` was due to a crash from the `EAMBINH` 1.2 test, where `overridden` happened to be `TraitMethodHandle`, which doesn't have `method_ast`. I traced back `impl.site` vs `impl.method_ast.site`, and found that `impl.site` always corresponded to the site of the method body AST, so I just refactored to use `impl.site` directly instead.